hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -759,6 +816,40 @@ private static boolean hiveEngineEnabled(TableMetadata metadata, Configuration c
         ConfigProperties.ENGINE_HIVE_ENABLED, TableProperties.ENGINE_HIVE_ENABLED_DEFAULT);
   }
 
+  private Pair<Long, LockState> findLock(String agentInfo) throws TException, InterruptedException {
+    // Search for the locks using HMSClient.showLocks identified by the agentInfo
+    ShowLocksRequest showLocksRequest = new ShowLocksRequest();
+    showLocksRequest.setDbname(database);
+    showLocksRequest.setTablename(tableName);
+    ShowLocksResponse response = metaClients.run(client -> client.showLocks(showLocksRequest));
+    for (ShowLocksResponseElement lock : response.getLocks()) {
```

Comment:
Not familiar with this. But excuse, is it possible to have multiple agentInfo locked in it?

Reply:
If there is a concurrent commit with its own lock, then we need to find our own